### PR TITLE
feat(module: date-picker, time-picker): support custom suffix icon

### PR DESCRIPTION
**Pull request by danranVm**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/core/config/config.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export interface NzConfig {
   codeEditor?: CodeEditorConfig;
   collapse?: CollapseConfig;
   collapsePanel?: CollapsePanelConfig;
+  date?: DateConfig;
```
Comment (wenqi73) [inline components/core/config/config.ts:29]:
```suggestion
  datePicker?: DateConfig;
```

File: components/core/config/config.ts [hunk 2]
```diff
@@ -122,6 +123,11 @@ export interface CollapsePanelConfig {
   nzShowArrow?: boolean;
 }
 
+export interface DateConfig {
```
Comment (wenqi73) [inline components/core/config/config.ts:126]:
```suggestion
export interface DatePickerConfig {
```

**Comment by wenqi73** [comment]
@danranVm time-picker 再加个测试吧。

**Comment by vthinkxie** [comment]
rebase needed

File: components/core/config/config.ts [hunk 3]
```diff
@@ -230,6 +236,7 @@ export interface TimePickerConfig {
   nzSecondStep?: number;
   nzPopupClassName?: string;
   nzUse12Hours?: string;
+  nzSuffixIcon?: string | TemplateRef<NzSafeAny>;
```
Comment (vthinkxie) [inline components/core/config/config.ts:239]:
```suggestion
```

**Pull request merged**